src/fontra/views/editor/panel-reference-font.js

```diff
@@ -322,6 +323,51 @@ export default class ReferenceFontPanel extends Panel {
         cssString;
       this.editorController.canvasController.requestUpdate();
     });
+
+    this.initActions();
+  }
+
+  initActions() {
+    const topic = "0200-action-topics.reference-font";
+    registerAction(
+      "action.select-previous-reference-font",
+      {
+        topic,
+        titleKey: "menubar.view.select-previous-reference-font",
+        defaultShortCuts: [],
+      },
+      () => this.doSelectPreviousNextReferenceFont(true)
+    );
+
+    registerAction(
+      "action.select-next-reference-font",
+      {
+        topic,
+        titleKey: "menubar.view.select-next-reference-font",
+        defaultShortCuts: [],
+      },
+      () => this.doSelectPreviousNextReferenceFont(false)
+    );
+  }
+
+  async doSelectPreviousNextReferenceFont(selectPrevious) {
+    const listLength = this.filesUIList.items.length;
+    if (!listLength) {
+      return;
+    }
+
+    const index = this.filesUIList.getSelectedItemIndex();
+
+    const newIndex = isNaN(index)
```

Comment:
```js
  const newIndex = modulo(index + (selectPrevious ? -1 : 1), listLength);
```
`modulo()` from utils.js.

Reply:
`doSelectPreviousNextSource()` uses JS's `%` modulo, and adds numSources to make it work for negative numbers. Our `modulo()` function behaves like Python's `%` and does the right thing. `doSelectPreviousNextSource()` should probably follow the same pattern.

```js
$ node
> 14 % 6
2
> -14 % 6
-2
> 
```

```python
$ python
>>> 14 % 6
2
>>> -14 % 6
4
>>> 
```

Reply:
Oh, that's impressive. Thanks. I add it as suggested. And for the doSelectPreviousNextSource I created a new issue. I am closing this as resolved.